providers/celery/src/airflow/providers/celery/executors/celery_executor_utils.py

```diff
@@ -125,21 +130,54 @@ def on_celery_import_modules(*args, **kwargs):
         import kubernetes.client  # noqa: F401
 
 
-@app.task
-def execute_command(command_to_exec: CommandType) -> None:
-    """Execute command."""
-    dag_id, task_id = BaseExecutor.validate_airflow_tasks_run_command(command_to_exec)
+# Once Celery5 is out of beta, we can pass `pydantic=True` to the decorator and it will handle the validation
+# and deserialization for us
+@app.task(name="execute_workload")
+def execute_workload(input: str) -> None:
+    from pydantic import TypeAdapter
+
+    from airflow.configuration import conf
+    from airflow.executors import workloads
+    from airflow.sdk.execution_time.supervisor import supervise
+
+    decoder = TypeAdapter(workloads.All)
+    workload = decoder.validate_json(input)
```

Comment:
Should this rather be in a try block?
```
            try:
                workload = decoder.validate_json(input)
```

Reply:
I don't think so - we want it to fail so I don't know what we'd put in the except block other than raise

Reply:
Yeah doesnt add any value